### PR TITLE
storage: add pull and deletes plus tests to storageclass

### PR DESCRIPTION
**Pull request by klaskosk**
Added a PullClass function to the storage package; added Delete, DeleteAndWait, and WaitUntilDeleted methods to ClassBuilder; and added unit tests for all of the functions in `storageclass.go`.

### CODE REVIEW

**Review by kononovn — CHANGES_REQUESTED** [review]
Looks good just few changes are needed.

**Comment by kononovn** [comment]
> @kononovn Updated the test, although there's no check for a nil client in `NewClassBuilder` since the `apiClient` field hasn't been narrowed so it would be redundant with `validate`

I know just to make it more consistent can you add to NewBuilder following condition?
https://github.com/openshift-kni/eco-goinfra/blob/30c5d20d5b39439e9d23a9740f2b5f6eef4ac9ab/pkg/assisted/agentclusterinstall.go#L51

**Review by kononovn — CHANGES_REQUESTED** [review]
please check comment above.

**Review by kononovn — DISMISSED** [review]
/lgtm

File: pkg/storage/storageclass.go [hunk 1]
```diff
@@ -217,6 +261,52 @@ func (builder *ClassBuilder) Delete() error {
...
 }
 
+// DeleteAndWait deletes the StorageClass and waits up to timeout until it has been removed.
+func (builder *ClassBuilder) DeleteAndWait(timeout time.Duration) error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof(
+		"Deleting StorageClass %s and waiting up to %s until it is removed", timeout, builder.Definition.Name)
```
Comment (trewest) [inline pkg/storage/storageclass.go:271]: Looks like the variables are ordered incorrectly

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**